File: Tan Khoa/capdoi.cpp
#include <bits/stdc++.h>

using namespace std;

#define long long int
#define nln '\n'

signed main() {
	cin.tie(0)->sync_with_stdio(0);
	int n, x;
	cin >> n >> x;
	vector<int> arr(n);
	for (auto &v : arr)
		cin >> v;

	map<int, int> cnt;
	for (auto v : arr)
		cnt[v]++;

	sort(arr.begin(), arr.end());

	int left = 0, right = n-1, ans = 0;
	bool stop = false;
	while (left < right && !stop) {
		if (arr[left] + arr[right] > x) {
			right -= 1;
		} else if (arr[left] + arr[right] < x) {
			left += 1;
		} else {
			if (arr[left] == arr[right]) {
				ans += cnt[arr[left]]*(cnt[arr[left]]-1)/2;
				stop = true;
			} else
				ans += cnt[arr[right]];
			left += 1;
		}
		if (stop == true)
			break;
	}

	cout << ans << nln;

	return 0;
}

File: Tan Khoa/capdoi.py
import sys 
from collections import defaultdict

input = sys.stdin.readline # faster

n, x = map(int, input().split())
arr = list(map(int, input().split()))

cnt = defaultdict(lambda : 0)
for v in arr:
	cnt[v] += 1

arr.sort()

l = 0; r = n-1
ans = 0

while l < r:
	if arr[l] + arr[r] < x:
		l += 1
	elif arr[l] + arr[r] > x:
		r -= 1
	else:
		if arr[l] != arr[r]:
			ans += cnt[arr[r]]
			l += 1
		else:
			ans += (cnt[arr[l]]-1)*cnt[arr[l]]//2
			break

print(ans)


